syntax = "proto3";

package mealprep;

message Recipe {
  string recipe_id = 1;
  string title = 2;
  string description = 3;
  repeated Ingredient ingredients = 4;
  repeated InstructionSection instructions = 5;
  int32 prep_time_minutes = 6;
  int32 cook_time_minutes = 7;
  NutritionProfile nutrition = 8;
  int32 servings = 9;
  string serving_size = 10;
  repeated string citations = 11;
}

message Ingredient {
  string name = 1;
  float quantity = 2;
  string unit = 3;
  string notes = 4;
}

message InstructionSection {
  string section_name = 1;
  repeated string steps = 2;
}

message NutritionProfile {
  int32 calories = 1;
  float protein_grams = 2;
  float carbs_grams = 3;
  float fat_grams = 4;
  float fiber_grams = 5;
  float sugar_grams = 6;
  float sodium_mg = 7;
}
